virtcontainers/device/drivers/ephemeral.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017-2018 Intel Corporation
+// Copyright (c) 2018 Huawei Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package drivers
+
+import (
+	"github.com/kata-containers/runtime/virtcontainers/device/api"
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+)
+
+// GenericDevice refers to a device that is neither a VFIO device or block device.
+type EphemeralDevice struct {
```

Comment:
I don't see where this EphemeralDevice is been used, I think this file is unnecessary, right?

Reply:
You are correct, this has not been used and will be removed from final PR. I was just trying to model it like `BlockDevice` but unlike `BlockDevice` I didn't find a way to use it anywhere since we don't end up actually passing any device to the container.

Reply:
I'll mark this as "need change" in case we forget and merge it in accidently.